Wangscape/noise/Reseedable.h

```diff
@@ -58,8 +58,8 @@ struct Reseedable final
 template<typename InputIt>
 inline Reseedable Reseedable::terrace(InputIt first, InputIt last, bool inverted)
 {
-    static_assert(std::is_same<std::iterator_traits<InputIt>::iterator_category::input_iterator_tag,
-                  std::input_iterator_tag>::value,
+    static_assert(std::is_same<typename std::iterator_traits<InputIt>::iterator_category,
```

Comment:
Right, I thought about it, but didn't check completely before. I'll change it to `std::is_base_of`.